modelling/src/main/java/org/axonframework/modelling/command/AnnotationCommandTargetResolver.java

```diff
@@ -70,44 +89,101 @@ public VersionedAggregateIdentifier resolveTarget(CommandMessage<?> command) {
-                ensureAccessible(m);
-                return Optional.ofNullable(m.invoke(command.getPayload())).map(Object::toString).orElse(null);
-            }
-        }
-        for (Field f : fieldsOf(command.getPayloadType())) {
-            if (f.isAnnotationPresent(TargetAggregateIdentifier.class)) {
-                return Optional.ofNullable(getFieldValue(f, command.getPayload())).map(Object::toString).orElse(null);
-            }
-        }
-        return null;
-    }
+	private String findIdentifier(Message<?> command) throws InvocationTargetException, IllegalAccessException {
+		return Optional.ofNullable(invokeAnnotated(command, identifierAnnotation)).map(Object::toString).orElse(null);
+	}
 
-    private Long findVersion(CommandMessage<?> command) throws InvocationTargetException, IllegalAccessException {
-        for (Method m : methodsOf(command.getPayloadType())) {
-            if (AnnotationUtils.isAnnotationPresent(m, TargetAggregateVersion.class)) {
-                ensureAccessible(m);
-                return asLong(m.invoke(command.getPayload()));
-            }
-        }
-        for (Field f : fieldsOf(command.getPayloadType())) {
-            if (AnnotationUtils.isAnnotationPresent(f, TargetAggregateVersion.class)) {
-                return asLong(getFieldValue(f, command.getPayload()));
-            }
-        }
-        return null;
-    }
+	private Long findVersion(Message<?> command) throws InvocationTargetException, IllegalAccessException {
+		return asLong(invokeAnnotated(command, versionAnnotation));
+	}
 
-    private Long asLong(Object fieldValue) {
-        if (fieldValue == null) {
-            return null;
-        } else if (Number.class.isInstance(fieldValue)) {
-            return ((Number) fieldValue).longValue();
-        } else {
-            return Long.parseLong(fieldValue.toString());
-        }
-    }
-}
+	private static Object invokeAnnotated(Message<?> command, Class<? extends Annotation> annotation)
+			throws InvocationTargetException, IllegalAccessException {
+		for (Method m : methodsOf(command.getPayloadType())) {
+			if (AnnotationUtils.isAnnotationPresent(m, annotation)) {
+				ensureAccessible(m);
+				return m.invoke(command.getPayload());
+			}
+		}
+		for (Field f : fieldsOf(command.getPayloadType())) {
+			if (AnnotationUtils.isAnnotationPresent(f, annotation)) {
+				return getFieldValue(f, command.getPayload());
+			}
+		}
+		return null;
+	}
+
+	private Long asLong(Object fieldValue) {
+		if (fieldValue == null) {
+			return null;
+		} else if (Number.class.isInstance(fieldValue)) {
+			return ((Number) fieldValue).longValue();
+		} else {
+			return Long.parseLong(fieldValue.toString());
+		}
+	}
+
+	@Override
+	public String toString() {
+		return "AnnotationCommandTargetResolver [identifierAnnotation=" + identifierAnnotation + ", versionAnnotation="
+				+ versionAnnotation + "]";
+	}
+
+	public static final class Builder {
+		private AnnotationCommandTargetResolver resolver;
```

Comment:
I'd thus suggest to replace the `resolver` field here for the `identifierAnnotation` and `versionAnnotation` fields, setting the default `TargetAggregateIdentifier.class` and `TargetAggregateVersion.class` on them as well.